### PR TITLE
fix new-app errors

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/3190

@csrwng ptal

### CODE REVIEW

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func (c *AppConfig) resolve(components app.ComponentReferences) error {
 			continue
 		case input.ExpectToBuild && !input.Match.Builder:
```
Comment (csrwng) [inline pkg/generate/app/cmd/newapp.go]: the condition should also include !input.Uses.IsDockerBuild():
```
case input.ExpectToBuild && !input.Match.Builder && !input.Uses.IsDockerBuild():
```

Comment (deads2k) [inline pkg/generate/app/cmd/newapp.go]: done

**Comment by csrwng** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[test]

**Pull request merged**